pkg/controller/validate_config.go

```diff
@@ -104,3 +111,78 @@ func (c *ValidateConfigHandler) Handle(ctx context.Context) {
 	ctx = CtxClusterStatus.WithValue(ctx, currentStatus)
 	c.next.Handle(ctx)
 }
+
+func (c *ValidateConfigHandler) getAvailableVersions(ctx context.Context, graph updates.UpdateGraph, version v1alpha1.SpiceDBVersion, datastore string) []v1alpha1.SpiceDBVersion {
+	logger := logr.FromContextOrDiscard(ctx)
+	source, err := graph.SourceForChannel(version.Channel)
+	if err != nil {
+		logger.V(4).Error(err, "no source found for channel %q, can't compute available versions", version.Channel)
+	}
+
+	availableVersions := make([]v1alpha1.SpiceDBVersion, 0)
+	nextDirect := source.NextDirect(version.Name)
```

Comment:
`NextDirect` vs `Next` also might do better with slightly more descriptive names

Reply:
I agree, but I'm drawing a blank. Any suggestions?

Reply:
`NextVersion` and `NextVersionWithoutMigration`?